File: CrackingCodingInterview/sol1.4/main.py
import sys

class Solution(object):
    def check_permutation_palindrome(self, strr):
        single_ctr = 0
        arr = [False] * 256
        for ch in strr:
            if arr[ord(ch)] == False:
                single_ctr += 1
                arr[ord(ch)] = True;
            else:
                single_ctr -= 1
                arr[ord(ch)] = False
        if single_ctr == 0 or single_ctr == 1:
            return True
        else:
            return False

def main():
    ret = Solution().check_permutation_palindrome(sys.argv[1]);
    if ret == True:
        print "It is permutation palindrome"
    else:
        print "Its not a permutation palindrome"

if __name__ == "__main__":
    main()

File: CrackingCodingInterview/sol1.4/main.cpp
#include <iostream>
#include <string>
using namespace std;
class Solution {
    public:
        bool check_permutation_palindrome(string str) {
            int single_ctr = 0;
            bool arr[256] = {false};

            for (auto &x : str) {
                if (arr[x] == false) {
                    single_ctr++; 
                    arr[x] = true;
                }
                else {
                    single_ctr--; 
                    arr[x] = false;
                }
            }
            if (single_ctr == 0  || single_ctr == 1)
                return true;
            else
                return false;
        }
};

int main() {

    string str;
    cout << "Enter string to be checked: ";
    cin >> str;
    bool ret = Solution().check_permutation_palindrome(str);
    if (ret == true)
        cout << "String is permutation palindrome";
    else
        cout << "String is not permutation palindrome";
    return 0;
}
